yt_idefix/_io/h5_io.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+import os
+
+import numpy as np
+
+from yt.utilities.on_demand_imports import _h5py as h5py
+
+from .commons import Coordinates, Shape, get_native_coordinates_from_cartesian
+
+KNOWN_GEOMETRIES: dict[int, str] = {
+    0: "cartesian",
+    1: "polar",
+    2: "spherical",
+    3: "cylindrical",
+}
+
+
+def read_grid_coordinates(
+    filename: str | os.PathLike[str],
+    *,
+    geometry: str | None = None,
+) -> Coordinates:
+    # Return cell edges coordinates
+    fh = h5py.File(filename, "r")
+    if geometry not in (valid_geometries := tuple(KNOWN_GEOMETRIES.values())):
+        raise ValueError(
+            f"Got unknown geometry {geometry!r}, expected one of {valid_geometries}"
+        )
+
+    nodesX = fh["/node_coords/X"].astype("=f8")
+    nodesY = fh["/node_coords/Y"].astype("=f8")
+    nodesZ = fh["/node_coords/Z"].astype("=f8")
+    dimensions = len(np.array(nodesX).shape)
+    shape = Shape(
+        *np.array(nodesX).shape
+    )  # this is reversed compared the vtk implementation in vtk_io.py
+    coords: list[np.ndarray] = []
+    # now assuming that fh is positioned at the end of metadata
+    if geometry in ("cartesian", "cylindrical") or (
+        geometry in ("polar", "spherical") and dimensions == 1
+    ):
+        if dimensions == 1:
+            nodesX = np.array(nodesX)
+            nodesY = np.array(
+                [0.0, 1.0]
+            )  # Default is assumed and not parsed from pluto.ini/grid.out (not present in data)
```

Comment:
long inline comments tend to hurt readability when formatted with black. The way to work around it is to put the comment on a separate line